std/exception.d

```diff
@@ -2148,3 +2148,65 @@ pure nothrow @safe unittest
 
     auto infSlice = infinite[0 .. $]; // this would throw otherwise
 }
+
+
+/++
+    Convenience mixin for trivially sub-classing exceptions
+
+    Even trivially sub-classing an exception involves writing boilerplate code
+    for the constructor to: 1) correctly pass in the source file and line number
+    the exception was thrown from; 2) be usable with $(LREF enforce) which
+    expects exception constructors to take arguments in a fixed order. This
+    mixin provides that boilerplate code.
+
+    Note however that you need to mark the $(B mixin) line with at least a
+    minimal (i.e. just $(B ///) DDoc comment if you want the mixed-in
+    constructors to be documented in the newly created Exception subclass.
+ +/
+mixin template basicExceptionCtors()
+{
+    /++
+        Params:
+            msg  = The message for the exception.
+            file = The file where the exception occurred.
+            line = The line number where the exception occurred.
+            next = The previous exception in the chain of exceptions, if any.
+    +/
+    this(string msg, string file = __FILE__, size_t line = __LINE__,
+         Throwable next = null) @safe pure nothrow
+    {
+        super(msg, file, line, next);
+    }
+
+    /++
+        Params:
+            msg  = The message for the exception.
+            next = The previous exception in the chain of exceptions.
+            file = The file where the exception occurred.
+            line = The line number where the exception occurred.
+    +/
+    this(string msg, Throwable next, string file = __FILE__,
+         size_t line = __LINE__) @safe pure nothrow
+    {
+        super(msg, file, line, next);
+    }
+}
+
+///
+pure nothrow @safe unittest
+{
+    class MeaCulpa: Exception
+    {
+        ///
+        mixin basicExceptionCtors;
+    }
+
+    try
+        throw new MeaCulpa("test");
+    catch (Exception e) // Bugzilla 15467
+    {
+        assert(e.msg == "test");
```

Comment:
From your extra `unittest` above I gather that the new `message` is a function but can be called as a property without the `()`, but this is not obvious from [the diff](https://github.com/WalterBright/druntime/commit/2fbe4a8b393d73c468721da4c550987d9a2c8d14) since I don't see any `@property` there. Or am I mistaken in thinking that you need to have an `@property` to call a function without parantheses? Please clarify.